Mathlib/Combinatorics/SimpleGraph/Operations.lean

```diff
@@ -0,0 +1,170 @@
+/-
+Copyright (c) 2024 Jeremy Tan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jeremy Tan
+-/
+import Mathlib.Combinatorics.SimpleGraph.Basic
+
+/-!
+# Local graph operations
+
+This file defines some single-graph operations that modify a finite number of vertices
+and proves basic theorems about them. When the graph itself has a finite number of vertices
+we also prove theorems about the number of edges in the modified graphs.
+
+## Main definitions
+
+* `G.replaceVertex s t` is `G` with `t` replaced by a copy of `s`,
+  removing the `s-t` edge if present.
+* `G.addEdge s t` is `G` with the `s-t` edge added, if that is a valid edge.
+-/
+
+
+open Finset
+
+namespace SimpleGraph
+
+variable {V : Type*} [DecidableEq V] (G : SimpleGraph V) (s t : V)
+
+section ReplaceVertex
+
+/-- The graph formed by forgetting `t`'s neighbours and instead giving it those of `s`. The `s-t`
+edge is removed if present. -/
+abbrev replaceVertex : SimpleGraph V where
```

Comment:
Why should this be an `abbrev`, not `def`?

Reply:
When I first wrote this code I had `def`, but then someone told me to use `abbrev` instead so I wouldn't have to write e.g. `simp [replaceVertex]` to close a goal all the time.